### PR TITLE
First Steps into making Gradle Catalogs

### PR DESCRIPTION
**Pull request by JBassett**
## Summary

Gradle Catalogs!

## Screenshots

N/A

## Link to pull request in Documentation repository

N/A

### CODE REVIEW

**Comment by JBassett** [comment]
I realize these might not be named exactly how we want, if anyone sees anything they want to see changed please add a review and I'll refactor it.

**Comment by jpelgrom** [comment]
Two things that come to mind, but maybe you intend do it in a later stage:
- shouldn't this also include the [plugins](https://developer.android.com/build/migrate-to-catalogs#migrate-plugins) (currently only ktlint is added, but we have a lot more)
- can some dependencies be bundled like Room, ExoPlayer, Compose? (looking at [this article it seems relatively easy](https://proandroiddev.com/gradle-version-catalogs-for-an-awesome-dependency-management-f2ba700ff894))

**Comment by JBassett** [comment]
> Two things that come to mind, but maybe you intend do it in a later stage:
> 
> * shouldn't this also include the [plugins](https://developer.android.com/build/migrate-to-catalogs#migrate-plugins) (currently only ktlint is added, but we have a lot more)
> * can some dependencies be bundled like Room, ExoPlayer, Compose? (looking at [this article it seems relatively easy](https://proandroiddev.com/gradle-version-catalogs-for-an-awesome-dependency-management-f2ba700ff894))

So I tried to keep it to as much of a 1 to 1 mapping of existing to new, due to differences in how plugins are declared currently it wasn't as easy and I wanted to do separately.  As far as the bundling I think handling them separately will make the PRs cleaner and easier to read.

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+biometric = { module = "androidx.biometric:biometric", version.ref = "biometric" }
+blurView = { module = "com.github.Dimezis:BlurView", version.ref = "blurView" }
+changeLog = { module = "com.github.AppDevNext:ChangeLog", version.ref = "changeLog" }
+community-material-typeface = { module = "com.mikepenz:community-material-typeface", version.ref = "community-material-typeface" }
+compose-bom = { module = "androidx.compose:compose-bom", version.ref = "compose-bom" }
+compose-animation = { module = "androidx.compose.animation:animation" }
+compose-foundation = { module = "androidx.compose.foundation:foundation" }
+compose-material = { module = "androidx.compose.material:material" }
+compose-material-iconsCore = { module = "androidx.compose.material:material-icons-core" }
+compose-material-iconsExtended = { module = "androidx.compose.material:material-icons-extended" }
```
Comment (jpelgrom) [inline gradle/libs.versions.toml:99]: Why switch to camel case at the end?

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+androidx-room-ktx = { module = "androidx.room:room-ktx", version.ref = "roomKtx" }
+androidx-room-runtime = { module = "androidx.room:room-runtime", version.ref = "roomRuntime" }
+androidx-lifecycle-viewmodel-ktx = { module = "androidx.lifecycle:lifecycle-viewmodel-ktx", version.ref = "lifecycleViewmodelKtx" }
+androidx-tiles-material = { module = "androidx.wear.tiles:tiles-material", version.ref = "tilesMaterial" }
+androidx-tiles = { module = "androidx.wear.tiles:tiles", version.ref = "tiles" }
+androidx-watchface-complications-data-source-ktx = { module = "androidx.wear.watchface:watchface-complications-data-source-ktx", version.ref = "watchfaceComplicationsDataSourceKtx" }
+androidx-work-runtime-ktx = { module = "androidx.work:work-runtime-ktx", version.ref = "workRuntimeKtx" }
+app-car = { module = "androidx.car.app:app", version.ref = "car-app" }
+app-automotive = { module = "androidx.car.app:app-automotive", version.ref = "car-app" }
+app-projected = { module = "androidx.car.app:app-projected", version.ref = "car-app" }
```
Comment (jpelgrom) [inline gradle/libs.versions.toml:88]: I'd keep car in the name here, for example "libs.app.projected" isn't clearly car related.

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+compose-bom = { module = "androidx.compose:compose-bom", version.ref = "compose-bom" }
+compose-animation = { module = "androidx.compose.animation:animation" }
+compose-foundation = { module = "androidx.compose.foundation:foundation" }
+compose-material = { module = "androidx.compose.material:material" }
+compose-material-iconsCore = { module = "androidx.compose.material:material-icons-core" }
+compose-material-iconsExtended = { module = "androidx.compose.material:material-icons-extended" }
+compose-runtime = { module = "androidx.compose.runtime:runtime" }
+compose-ui = { module = "androidx.compose.ui:ui" }
+compose-uiTooling = { module = "androidx.compose.ui:ui-tooling" }
+compiler = { module = "androidx.compose.compiler:compiler", version.ref = "compose-compiler" }
```
Comment (jpelgrom) [inline gradle/libs.versions.toml:103]: Can we add Compose to the name? Compiler is very general

**Comment by jpelgrom** [comment]
> So I tried to keep it to as much of a 1 to 1 mapping of existing to new, due to differences in how plugins are declared currently it wasn't as easy and I wanted to do separately. As far as the bundling I think handling them separately will make the PRs cleaner and easier to read.

Sounds like you've thought about it, good plan.

**Review by jpelgrom — COMMENTED** [review]
Names that I noticed while scanning the list / weren't obvious to me until I looked at the toml file

File: gradle/libs.versions.toml [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+playPublisher = "3.8.4"
+preference-ktx = "1.2.0"
+recyclerview = "1.3.1"
+lifecycle-runtime-ktx = "2.6.1"
+reorderable = "0.9.6"
+retrofit = "2.9.0"
+roomKtx = "2.5.2"
+roomRuntime = "2.5.2"
+tilesMaterial = "1.1.0"
+tiles = "1.1.0"
```
Comment (dshokouhi) [inline gradle/libs.versions.toml:53]: shoudl we consider adding `wear-` to the wear specific libraries to make them more easily identifiable?

File: gradle/libs.versions.toml (hunk 1 above)
Comment (JBassett) [reply gradle/libs.versions.toml]: Fixed

File: gradle/libs.versions.toml (hunk 2 above)
Comment (JBassett) [reply gradle/libs.versions.toml]: Fixed

File: gradle/libs.versions.toml (hunk 3 above)
Comment (JBassett) [reply gradle/libs.versions.toml]: Fixed

File: gradle/libs.versions.toml (hunk 4 above)
Comment (JBassett) [reply gradle/libs.versions.toml]: Fixed

File: gradle/libs.versions.toml [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+picasso = { module = "com.squareup.picasso:picasso", version.ref = "picasso" }
+play-services-wearable = { module = "com.google.android.gms:play-services-wearable", version.ref = "play-services-wearable" }
+preference-ktx = { module = "androidx.preference:preference-ktx", version.ref = "preference-ktx" }
+recyclerview = { module = "androidx.recyclerview:recyclerview", version.ref = "recyclerview" }
+lifecycle-runtime-ktx = { module = "androidx.lifecycle:lifecycle-runtime-ktx", version.ref = "lifecycle-runtime-ktx" }
+reorderable = { module = "org.burnoutcrew.composereorderable:reorderable", version.ref = "reorderable" }
+retrofit = { module = "com.squareup.retrofit2:retrofit", version.ref = "retrofit" }
+wear = { module = "androidx.wear:wear", version.ref = "wear" }
+wear-androidx-tiles-material = { module = "androidx.wear.tiles:tiles-material", version.ref = "wear-tiles" }
+wear-androidx-tiles = { module = "androidx.wear.tiles:tiles", version.ref = "wear-tiles" }
```
Comment (jpelgrom) [inline gradle/libs.versions.toml:146]: androidx in the name feels unnecessary here, all other androidx libraries skip it

Comment (JBassett) [reply gradle/libs.versions.toml]: done

File: gradle/libs.versions.toml [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+accompanist-themeadapter-material = { module = "com.google.accompanist:accompanist-themeadapter-material", version.ref = "accompanist" }
+accompanist-systemuicontroller = { module = "com.google.accompanist:accompanist-systemuicontroller", version.ref = "accompanist" }
+activity-compose = { module = "androidx.activity:activity-compose", version.ref = "activity-compose" }
+activity-ktx = { module = "androidx.activity:activity-ktx", version.ref = "activity-compose" }
+android-beacon-library = { module = "org.altbeacon:android-beacon-library", version.ref = "androidBeaconLibrary" }
+android-plugin = { module = "com.android.tools.build:gradle", version.ref = "androidPlugin" }
+android-junit5 = { module = "de.mannodermaus.gradle.plugins:android-junit5", version.ref = "androidJunit5" }
+androidx-compose-foundation = { module = "androidx.wear.compose:compose-foundation", version.ref = "composeFoundation" }
+androidx-compose-material = { module = "androidx.wear.compose:compose-material", version.ref = "composeFoundation" }
+androidx-compose-navigation = { module = "androidx.wear.compose:compose-navigation", version.ref = "composeFoundation" }
```
Comment (jpelgrom) [inline gradle/libs.versions.toml:75]: These names aren't helpful, what would the different be between compose-foundation and androidx-compose-foundation? Suggestion: make it something like wear-compose-foundation.

File: gradle/libs.versions.toml [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+material = { module = "com.google.android.material:material", version.ref = "material" }
+play-publisher = { module = "com.github.triplet.gradle:play-publisher", version.ref = "playPublisher" }
+play-services-threadnetwork = { module = "com.google.android.gms:play-services-threadnetwork", version.ref = "play-services-threadnetwork" }
+play-services-home = { module = "com.google.android.gms:play-services-home", version.ref = "play-services-home" }
+play-services-location = { module = "com.google.android.gms:play-services-location", version.ref = "play-services-location" }
+picasso = { module = "com.squareup.picasso:picasso", version.ref = "picasso" }
+play-services-wearable = { module = "com.google.android.gms:play-services-wearable", version.ref = "play-services-wearable" }
+preference-ktx = { module = "androidx.preference:preference-ktx", version.ref = "preference-ktx" }
+recyclerview = { module = "androidx.recyclerview:recyclerview", version.ref = "recyclerview" }
+lifecycle-runtime-ktx = { module = "androidx.lifecycle:lifecycle-runtime-ktx", version.ref = "lifecycle-runtime-ktx" }
```
Comment (jpelgrom) [inline gradle/libs.versions.toml:141]: Out of alphabetical order (move up)

File: gradle/libs.versions.toml [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+reorderable = { module = "org.burnoutcrew.composereorderable:reorderable", version.ref = "reorderable" }
+retrofit = { module = "com.squareup.retrofit2:retrofit", version.ref = "retrofit" }
+wear = { module = "androidx.wear:wear", version.ref = "wear" }
+wear-phone-interactions = { module = "androidx.wear:wear-phone-interactions", version.ref = "wearPhoneInteractions" }
+wear-input = { module = "androidx.wear:wear-input", version.ref = "wearInput" }
+wear-tiles-material = { module = "androidx.wear.tiles:tiles-material", version.ref = "wear-tiles" }
+wear-tiles = { module = "androidx.wear.tiles:tiles", version.ref = "wear-tiles" }
+webkit = { module = "androidx.webkit:webkit", version.ref = "webkit" }
+wear-remote-interactions = { module = "androidx.wear:wear-remote-interactions", version.ref = "wear-remote-interactions" }
+sentry-android = { module = "io.sentry:sentry-android", version.ref = "sentry-android" }
```
Comment (jpelgrom) [inline gradle/libs.versions.toml:151]: Out of alphabetical order (move up) x2

File: gradle/libs.versions.toml [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+composeFoundation = "1.1.2"
+converterJackson = "2.9.0"
+coreKtx = "1.10.1"
+cronet-embedded = "113.5672.61"
+emojiJava = "5.1.1"
+exoplayer = "2.19.0"
+firebase-bom = "32.2.0"
+firebaseAppdistributionGradle = "4.0.0"
+fragment-ktx = "1.6.1"
+constraintlayout = "2.1.4"
```
Comment (jpelgrom) [inline gradle/libs.versions.toml:24]: Out of alphabetical order (move up)

File: gradle/libs.versions.toml [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+hilt = "2.47"
+iconics = "5.4.0"
+jackson-module-kotlin = "2.13.5"
+kotlinx-coroutines = "1.7.2"
+kotlin = "1.8.22"
+ktlint = "11.5.0"
+lifecycleViewmodelKtx = "2.6.1"
+loggingInterceptor = "4.11.0"
+navigation-compose = "2.6.0"
+okhttp = "4.11.0"
```
Comment (jpelgrom) [inline gradle/libs.versions.toml:37]: Out of alphabetical order (move down)

File: gradle/libs.versions.toml [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+material = "1.9.0"
+play-services-threadnetwork = "16.0.0"
+play-services-home = "16.0.0"
+play-services-location = "21.0.1"
+picasso = "2.8"
```
Comment (jpelgrom) [inline gradle/libs.versions.toml:42]: Out of alphabetical order (move up)

File: gradle/libs.versions.toml [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+play-services-wearable = "18.0.0"
+playPublisher = "3.8.4"
+preference-ktx = "1.2.0"
+recyclerview = "1.3.1"
+lifecycle-runtime-ktx = "2.6.1"
```
Comment (jpelgrom) [inline gradle/libs.versions.toml:47]: Out of alphabetical order (move up)

File: gradle/libs.versions.toml [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+roomKtx = "2.5.2"
+roomRuntime = "2.5.2"
+watchfaceComplicationsDataSourceKtx = "1.1.1"
+wear = "1.2.0"
+wear-tiles = "1.1.0"
+wearPhoneInteractions = "1.0.1"
+wearInput = "1.2.0-alpha02"
+webkit = "1.7.0"
+wear-remote-interactions = "1.0.0"
+sentry-android = "6.26.0"
```
Comment (jpelgrom) [inline gradle/libs.versions.toml:59]: Out of alphabetical order (move up) x2

**Review by jpelgrom — COMMENTED** [review]
One more name suggestion. I also noticed some dependencies that weren't sorted correctly when doing this a-z but feel free to disregard if you don't agree that would be a good way to sort these.

**Comment by JBassett** [comment]
> One more name suggestion. I also noticed some dependencies that weren't sorted correctly when doing this a-z but feel free to disregard if you don't agree that would be a good way to sort these.

Fixed the comments.  For now alphabetized will work, as I start grouping them we might sort in order of perceived importance, or by which module they are used.

**Review by jpelgrom — APPROVED** [review]
Let's merge this so more improvements can happen, we can always change more names later

**Pull request merged**